libsawtooth/src/receipt/store/diesel/mod.rs

```diff
@@ -956,6 +956,71 @@ pub mod tests {
         assert!(test_result.is_ok());
     }
 
+    /// Verify that all transaction receipts in a SQLite `DieselReceiptStore` are returned
+    /// in the correct order when added to the database one at a time
+    ///
+    /// 1. Create a new `DieselReceiptStore`
+    /// 2. Generate 20 transaction receipts and individually add the first 10 to the receipt store
+    /// 3. Add the remaining 10 transaction receipts at the same time in a vector
+    /// 4. Call `list_receipts_since` on the receipt store, passing in None to indicate all
+    ///    receipts should be listed
+    /// 5. Check that the receipts are returned in order and that various fields
+    ///    contain the expected values
+    /// 6. Check that the number of receipts returned is 20
+    #[test]
+    fn test_sqlite_list_receipts_order() {
+        let test_result = std::panic::catch_unwind(|| {
```

Comment:
Given that this test uses an in-memory SQLite instance, the `catch_unwind` isn't necessary.

Reply:
Fixed